core/appmodule/service.go

```diff
@@ -0,0 +1,24 @@
+package appmodule
+
+import (
+	"cosmossdk.io/core/blockinfo"
+	"cosmossdk.io/core/event"
+	"cosmossdk.io/core/gas"
+	"cosmossdk.io/core/store"
+)
+
+// Service bundles all the core services provided by a compliant runtime
+// implementation into a single services.
+//
+// NOTE: If new core services are provided, they shouldn't be added to this
+// interface which would be API breaking, but rather a cosmossdk.io/core/appmodule/v2.Service
+// should be created which extends this Service interface with new services.
+type Service interface {
+	store.KVStoreService
+	store.MemoryStoreService
+	store.TransientStoreService
+	event.Service
+	blockinfo.Service
+	gas.Service
+	RootInterModuleClient
```

Comment:
Is there value in creating these subservices separately?

In practice, for a smooth transition, I envision module developers (including us) just replacing `sdk.Context` with this bundle service. We can also remove a lot of `core/*/service.go` packages, and just have a single one.

Reply:
Yeah it might be possible to have one meta service. I think it is still nice to have clear separation of concerns and separate packages.